### ISSUE TITLE
Shutdown hook not called in init script on Amazon Linux

### ISSUE THREAD

**Issue by lox** [buildkite/agent#242]
I've noticed that my agents on Amazon Linux don't shutdown gracefully (after termination I still see them listed under the agents tab).

I'm following the advice on https://github.com/buildkite/agent/issues/234 and using the older init.d script. 

It looks like CentOS derived operating systems need a lock in `/var/lock/subsys`otherwise they will not shutdown services.

Seems like we need:

```bash
broken_lock_file_for_centos=/var/lock/subsys/buildkite-agent

# In the start block
touch "$broken_lock_file_for_centos"

# In the stop block
rm -f "$broken_lock_file_for_centos"
```

**Comment by lox** [comment]
https://forums.aws.amazon.com/thread.jspa?threadID=22513

**Comment by sj26** [comment]
I've pushed a fix for this in agent [version 2.1.5 to GitHub](https://github.com/buildkite/agent/releases/v2.1.5) and to the unstable package channels. Will give it a test and, if all good, promote it to stable.

**Comment by toolmantim** [comment]
Cool, this will be out in the next release
